pkg/controller/enterprisesearch/config.go

```diff
@@ -251,16 +253,32 @@ func associationConfig(c k8s.Client, ent entv1beta1.EnterpriseSearch) (*settings
 		return settings.NewCanonicalConfig(), nil
 	}
 
+	// origin of authenticated ent users setting changed starting 8.x
```

Comment:
should this comment be placed right above `if ver.IsSameOrAfter(version.From(8, 0, 0)) { `?

Reply:
I hesitated. My reasoning was to put the comment at the beginning of the section that handles this: we prepare the default value, get the version and then use the new format. But I agree that it can be weird when we read the code flow. Let's move the comment.